carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/FileManagementUtil.java

```diff
@@ -0,0 +1,286 @@
+
+                if (path.trim().equals("")) {
+                    zip.putNextEntry(new ZipEntry(folder.getName()));
+                } else {
+                    zip.putNextEntry(new ZipEntry(path + "/" + folder.getName()));
+                }
+                while ((len = inputStream.read(buf)) > 0) {
+                    zip.write(buf, 0, len);
+                }
+                inputStream.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private static void addFolderContentsToZip(String srcFolder, ZipOutputStream zip) {
+        File folder = new File(srcFolder);
+        String fileList[] = folder.list();
+        if (fileList != null) {
+            try {
+                int i = 0;
+                while (true) {
+                    if (fileList.length == i) {
+                        break;
+                    }
+                    if (new File(folder, fileList[i]).isDirectory()) {
+                        zip.putNextEntry(new ZipEntry(fileList[i] + "/"));
+                        zip.closeEntry();
+                    }
+                    addToZip("", srcFolder + "/" + fileList[i], zip);
+                    i++;
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private static void addFolderToZip(String path, String srcFolder, ZipOutputStream zip) {
+        File folder = new File(srcFolder);
+        String fileList[] = folder.list();
+        if (fileList != null) {
+            try {
+                int i = 0;
+                while (true) {
+                    if (fileList.length == i) {
+                        break;
+                    }
+                    String newPath = folder.getName();
+                    if (!path.equalsIgnoreCase("")) {
+                        newPath = path + "/" + newPath;
+                    }
+                    if (new File(folder, fileList[i]).isDirectory()) {
+                        zip.putNextEntry(new ZipEntry(newPath + "/" + fileList[i] + "/"));
+                        //					zip.closeEntry();
+                    }
+                    addToZip(newPath, srcFolder + "/" + fileList[i], zip);
+                    i++;
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Delete a given directory.
+     *
+     * @param dir directory to be deleted
+     * @throws IOException
+     */
+    public static void deleteDirectories(File dir) throws IOException {
+        File[] children = dir.listFiles();
+
+        if (children != null) {
+            for (File child : children) {
+                if (child != null) {
+                    String[] childrenOfChild = child.list();
+                    if(childrenOfChild != null) {
+                        if(childrenOfChild.length > 0) {
```

Comment:
This is the migrated code. It is refactored in https://github.com/wso2/carbon-maven-plugins/pull/8